civictechprojects/forms.py

```diff
@@ -114,35 +114,45 @@ def create_or_edit_event(request, event_id):
             raise PermissionDenied()
 
         is_created_original = event.is_created
+        fields_changed = False
         project_fields_changed = False
-        read_form_field_string(event, form, 'event_agenda')
-        read_form_field_string(event, form, 'event_description')
-        read_form_field_string(event, form, 'event_short_description')
+        fields_changed |= read_form_field_string(event, form, 'event_agenda')
+        fields_changed |= read_form_field_string(event, form, 'event_description')
+        fields_changed |= read_form_field_string(event, form, 'event_short_description')
         project_fields_changed |= read_form_field_string(event, form, 'event_name')
         project_fields_changed |= read_form_field_string(event, form, 'event_location')
-        read_form_field_string(event, form, 'event_rsvp_url')
-        read_form_field_string(event, form, 'event_live_id')
+        fields_changed |= read_form_field_string(event, form, 'event_rsvp_url')
+        fields_changed |= read_form_field_string(event, form, 'event_live_id')
         project_fields_changed |= read_form_field_string(event, form, 'event_organizers_text')
 
         project_fields_changed |= read_form_field_datetime(event, form, 'event_date_start')
         project_fields_changed |= read_form_field_datetime(event, form, 'event_date_end')
 
-        read_form_field_boolean(event, form, 'is_searchable')
-        read_form_field_boolean(event, form, 'is_created')
+        fields_changed |= read_form_field_boolean(event, form, 'is_searchable')
+        fields_changed |= read_form_field_boolean(event, form, 'is_created')
         project_fields_changed |= read_form_field_boolean(event, form, 'is_private')
 
         slug = form.data.get('event_slug')
         if slug is not None:
             slug = slug.lower()
-            slug_event = Event.get_by_slug(slug)
+            slug_event = Event.get_by_id_or_slug(slug)
             if slug_event and slug_event.id != event.id:
                 print('Could not change event {event} slug to {slug} because another event already has that slug: {existing_event}'.format(
                     event=event.__str__(),
                     slug=slug,
-                    existing_event=slug_event
+                    existing_event=slug_event.__str__()
                 ))
             else:
-                project_fields_changed |= read_form_field_string(event, form, 'event_slug', lambda str: str.lower())
+                # TODO: If we change back to an old slug, delete that NameRecord
```

Comment:
I'm not sure about this TODO. Here's my thinking. An event has slug A and is published. Later that's changed to slug B. Both are 'published' public slugs at some point. We retain knowledge of A even though the canonical slug is B so no links break. Even if the canonical slug is later changed back to A, wouldn't the same logic apply on retaining B?

It's a todo so obviously not blocking, just want to know what the reasoning is.

Reply:
Actually this did get done.  You're probably right in that retaining redundant slug records probably wouldn't hurt, but in that case I'd like us to stay consistent, in that we just make a Name Record every time a slug is set, and never delete them.  Does that sound good to you?

Reply:
Yeah, that makes a lot of sense to me. Since the business case was 'links don't break' it seems simpler to just keep name records as slugs are set. 👍